src/Dscp/Educator/Web/Student/Util.hs

```diff
@@ -0,0 +1,35 @@
+-- | Temporal mess of functions which didn't get into any other module.
+
+module Dscp.Educator.Web.Student.Util
+    ( assignTypeRaw
+    , verifySignedSubmission
+    , notImplemented
+    ) where
+
+import Control.Lens (Iso', iso)
+
+import Dscp.Core.Types (SignedSubmission)
+import qualified Dscp.Core.Types as Core
+import Dscp.Educator.Web.Student.Error (APIError (..))
+import Dscp.Educator.Web.Student.Types (IsFinal (..), Student)
+import Servant (err500, errBody)
+import qualified UnliftIO as UIO
+
+assignTypeRaw :: Iso' Core.AssignmentType IsFinal
+assignTypeRaw = iso forth back
+  where
+    back = \case
+        IsFinal False -> Core.Regular
+        IsFinal True  -> Core.CourseFinal
+    forth = \case
+        Core.Regular     -> IsFinal False
+        Core.CourseFinal -> IsFinal True
+
+-- | Checks that
+-- 1. 'SignedSubmission' is valid;
+-- 2. It was actually signed by a student.
+verifySignedSubmission :: Student -> SignedSubmission -> Either APIError ()
+verifySignedSubmission _ _ = pass  -- TODO [DSCP-141]
```

Comment:
No, DSCP-141 is about authenticating _every_ request, `SignedSubmission` is an independently verifiable entity, and moreover - the logic is already implemented in `Dscp.Educator.BlockVerification`, you just need to refactor it out a bit.